gamemode/modules/base/sv_gamemode_functions.lua

```diff
@@ -705,7 +705,9 @@ function GM:PlayerSpawn(ply)
 
     if ply.demotedWhileDead then
         ply.demotedWhileDead = nil
-        ply:changeTeam(GAMEMODE.DefaultTeam, true)
+        
+        local demoteTeam = hook.Call( "demoteTeam", nil, target ) or GAMEMODE.DefaultTeam
```

Comment:
Also, there's an error. It's not supposed to be `target`, but `ply`. I won't merge this until you tested this. Test everything, even the small changes.